astropy/visualization/wcsaxes/wcsapi.py

```diff
@@ -23,6 +24,94 @@
 
+UCD_COORD_META_MAPPING = {
+    "lon": {"coord_type": "longitude"},
+    "lat": {"coord_type": "latitude"},
+    "ra": {"coord_type": "longitude", "format_unit": u.hourangle},
+    "dec": {"coord_type": "latitude"},
+    "alt": {"coord_type": "longitude"},
+    "az": {"coord_type": "latitude"},
+    "long": {"coord_type": "longitude"},
+}
+
+CUSTOM_UCD_COORD_META_MAPPING = {
+    "pos.helioprojective.lon": {
+        "coord_wrap": 180.0 * u.deg,
+        "format_unit": u.arcsec,
+        "coord_type": "longitude",
+    },
+    "pos.helioprojective.lat": {"format_unit": u.arcsec, "coord_type": "latitude"},
+    "pos.heliographic.stonyhurst.lon": {
+        "coord_wrap": 180.0 * u.deg,
+        "format_unit": u.deg,
+        "coord_type": "longitude",
+    },
+    "pos.heliographic.stonyhurst.lat": {"format_unit": u.deg, "coord_type": "latitude"},
+    "pos.heliographic.carrington.lon": {
+        "coord_wrap": 360.0 * u.deg,
+        "format_unit": u.deg,
+        "coord_type": "longitude",
+    },
+    "pos.heliographic.carrington.lat": {"format_unit": u.deg, "coord_type": "latitude"},
+}
+
+
+class custom_ucd_wcscoord_mapping:
+    """
+    A context manager that makes it possible to temporarily add new UCD+ to WCS coordinate
+    plot metadata mappings.
+
+    Parameters
+    ----------
+    mapping : dict
+        A dictionary mapping a UCD to coordinate plot metadata
+
+    Examples
+    --------
+        >>> from matplotlib import pyplot as plt
+        >>> from astropy.visualization.wcsaxes.wcsapi import custom_ucd_wcscoord_mapping
+        >>> from astropy.wcs.wcsapi.fitswcs import custom_ctype_to_ucd_mapping
+        >>> wcs = WCS(naxis=1)
+        >>> wcs.wcs.ctype = ["eggs"]
+        >>> wcs.wcs.cunit = ["deg"]
+        >>> custom_mapping = {"eggs": "custom:pos.eggs"}
+        >>> with custom_ctype_to_ucd_mapping(custom_mapping):
+        ...     custom_meta = {
+        ...         "pos.eggs": {
+        ...             "coord_wrap": 360.0 * u.deg,
+        ...             "format_unit": u.arcsec,
+        ...             "coord_type": "longitude",
+        ...         }
+        ...     }
+        ...     with custom_ucd_wcscoord_mapping(custom_meta):
+        ...        fig = plt.figure()
+        ...        ax = fig.add_subplot(111, projection=wcs)
+        ...        ax.coords
+        <CoordinatesMap with 1 world coordinates:
+        <BLANKLINE>
+          index       aliases           type   unit  wrap format_unit visible
+                                                     deg
+          ----- -------------------- --------- ---- ----- ----------- -------
+              0 custom:pos.eggs eggs longitude  deg 360.0      arcsec     yes
+        <BLANKLINE>
+        >
+    """
+
+    def __init__(self, mapping):
+        self.mapping = {}
+        for k, v in mapping.items():
+            if k in CUSTOM_UCD_COORD_META_MAPPING:
+                raise ValueError(f"UCD metadata mapping {k} already exists.")
```

Comment:
Ideally we might want to allow existing entries to be overriden, but if this is too much work we could always keep as-is for now and consider it a future improvement to allow overriding.

Reply:
Patched in opt-in support for overwriting (with a kwarg)